iwyu.cc

```diff
@@ -2594,8 +2583,17 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     vector<const UsingDecl*> using_decls
         = FindInMultiMap(visitor_state_->using_declarations, decl);
     for (Each<const UsingDecl*> it(&using_decls); !it.AtEnd(); ++it) {
-      if (!(*it)->getDeclContext()->Encloses(using_context))
-        continue;
+      const DeclContext* using_decl_context = (*it)->getDeclContext();
+      if (!using_decl_context->Encloses(use_context)) {
```

Comment:
I find this a little hard to understand. On the one hand, you say that `Encloses` returns false for functions, but then on the other hand, you seem to say that it's never called for functions. Could the inner `if` be replaced by a `CHECK_`?

It seems wrong not to `continue` if it actually is a function. And if the prior predicate failed for functions, I'm not sure why you changed this.